src/color.rs

```diff
@@ -141,7 +141,7 @@ impl Color {
     ///
     /// FIXME(#2) Deprecated CSS2 System Colors are not supported yet.
     pub fn parse<'i, 't>(input: &mut Parser<'i, 't>) -> Result<Color, BasicParseError<'i>> {
-        let token = input.next()?;
+        let token = input.next()?.clone();
```

Comment:
Cloning is needed so that `input` is not still borrowed by the time we call `input.parse_nested_block` in the `Token::Function` case below. This is language limitation, non-lexical lifetimes will hopefully make this clone unnecessary. I’ll add a comment.

You’re right though that this clone makes the other one unnecessary, I’ll remove that.